dashboard/templatetags/cards.py

```diff
@@ -198,45 +256,56 @@ def card_statistics(child):
     stats = []
 
     changes = _diaperchange_statistics(child)
-    stats.append({
-        'type': 'duration',
-        'stat': changes['btwn_average'],
-        'title': _('Diaper change frequency')})
-
-    feedings = _feeding_statistics(child)
-    for item in feedings:
+    if changes:
         stats.append({
             'type': 'duration',
-            'stat': item['btwn_average'],
-            'title': item['title']})
+            'stat': changes['btwn_average'],
+            'title': _('Diaper change frequency')})
+
+    feedings = _feeding_statistics(child)
+    if feedings:
+        for item in feedings:
+            stats.append({
+                'type': 'duration',
+                'stat': item['btwn_average'],
+                'title': item['title']})
 
     naps = _nap_statistics(child)
-    stats.append({
-        'type': 'duration',
-        'stat': naps['average'],
-        'title': _('Average nap duration')})
-    stats.append({
-        'type': 'float',
-        'stat': naps['avg_per_day'],
-        'title': _('Average naps per day')})
+    if naps:
+        stats.append({
+            'type': 'duration',
+            'stat': naps['average'],
+            'title': _('Average nap duration')})
+        stats.append({
+            'type': 'float',
+            'stat': naps['avg_per_day'],
+            'title': _('Average naps per day')})
 
     sleep = _sleep_statistics(child)
-    stats.append({
-        'type': 'duration',
-        'stat': sleep['average'],
-        'title': _('Average sleep duration')})
-    stats.append({
-        'type': 'duration',
-        'stat': sleep['btwn_average'],
-        'title': _('Average awake duration')})
+    if sleep:
+        stats.append({
+            'type': 'duration',
+            'stat': sleep['average'],
+            'title': _('Average sleep duration')})
+        stats.append({
+            'type': 'duration',
+            'stat': sleep['btwn_average'],
+            'title': _('Average awake duration')})
 
     weight = _weight_statistics(child)
-    stats.append({
-        'type': 'float',
-        'stat': weight['change_weekly'],
-        'title': _('Weight change per week')})
+    if weight:
+        stats.append({
+            'type': 'float',
+            'stat': weight['change_weekly'],
+            'title': _('Weight change per week')})
+
+    empty = len(stats) == 0
```

Comment:
This _looks_ fine to me, but in my quick testing on your branch I never saw the statistics card -- it was always hidden even though there should have been statistics. Could you double check this functionality?

Reply:
You are right, there was an error in the template.